perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/DatabaseManagerBlImpl.java

```diff
@@ -113,7 +123,40 @@ protected void initialize() throws InternalErrorException {
 		}
 		log.debug("Initialize manager ends!");
 	}
-	
+
+	public java.sql.Array prepareSQLArrayOfNumbers(List<? extends PerunBean> perunBeans, PreparedStatement preparedStatement) throws SQLException, InternalErrorRuntimeException {
+		Connection connection = preparedStatement.getConnection().unwrap(Connection.class);
+		if(Compatibility.isOracle()) {
+			int[] arrayOfBeansIds = perunBeans.stream().mapToInt(PerunBean::getId).toArray();
+			try {
+				Method createOracleArrayMethod = connection.getClass().getMethod(nameOfOracleArrayMethod, String.class, Object.class);
+				createOracleArrayMethod.setAccessible(true);
+				return (java.sql.Array) createOracleArrayMethod.invoke(connection, AttributesManager.ORACLE_ARRAY_OF_NUMBERS, arrayOfBeansIds);
+			} catch (Exception ex) {
+				throw new InternalErrorRuntimeException("Can't access to method " + nameOfOracleArrayMethod, ex);
+			}
+		} else {
+			Integer[] arrayOfBeansIds = perunBeans.stream().map(PerunBean::getId).collect(Collectors.toList()).stream().toArray(Integer[]::new);
```

Comment:
I believe it can be shorter:
```suggestion
			Integer[] arrayOfBeansIds = perunBeans.stream().map(PerunBean::getId).toArray(Integer[]::new);
```